### DISCUSSION TITLE
Pixelfed follow request stuck

### DISCUSSION THREAD

**Discussion by chefblogger** [pixelfed/pixelfed#4939] [Support]
hi - i have a question and i hope someone could help me.

i have a raspberry pi 4 with an installed yunohost.

I am trying to follow my Pixelfed account via mastodon (also installed here on my raspi) (the instance only has me on it and login is closed).

i find my pixelfed via mastodon and when i click on follow - the follow request remains - i can only cancel it.

i have not set my pixelfed account to private - because you can see my pictures without logging in.

now my question - i can’t find a function in pixelfed where such a request can be made.

is this a bug or a hidden function that i can’t find?

thank you for your linktip / help for this

**Comment by TigerFox57** [comment]
I have a very similar issue. Even stranger, on the pixelfed server it says that the Mastodon user is following, and they can successfully receive messages from the Mastodon user as well as images, but direct message replies from the pixelfed user never reach the Mastodon user. On the Mastodon server, there's an option to cancel the follow, rather than unfollow, and the pixelfed user does not show uo in the list of those being followed. On the pixelfed server I can search for and follow the Mastodon user, but the Mastodon user never sees the follow.

No error messages show up anywhere in any of the pixelfed logs.

**Comment by TigerFox57** [comment]
Also, when I run php artisam test, there are 8 failed tests, but nothing strikes me as connected to my issue. Here are the failed tests.

  FAIL  Tests\Unit\Lexer\StatusLexerTest
  ✓ lexer extractor
  ⨯ autolink
  ✓ remote mention
  ✓ mention limit
  ✓ hashtag limit
  ✓ link limit

   FAIL  Tests\Unit\Lexer\UsernameTest
  ⨯ generic username
  ⨯ username with period
  ⨯ username with dash
  ⨯ username with underscore
  ⨯ multiple mentions
  ⨯ german umlats autolink
  ✓ german umlats extractor
  ⨯ german umlats webfinger autolink

**Comment by TigerFox57** [comment]
OK, I think I've fixed it. I changed the app environment setting in the .env file from local to production, and it started to work. It would help if I knew what I was doing...

**Reply by chefblogger** [reply]
congratulation for your solutions.

**Comment by chefblogger** [comment]
i checked the idea from @TigerFox57 with my raspbery pi but  my .env file the app_env setting is already "prodction"...

i've heard of others having a similar problem. some assume it has something to do with the ip being the same across all instances... 
i will definitely keep looking for a solution and tell you here when i know more

**Reply by TigerFox57** [reply]
This is interesting. Because When it started to work, it was working for the follow from a Mastodon account that was not hosted on my network, i.e mastodon.online. But it was still failing for follows from my locally hosted Mastodon account, running on a vm. I managed to fix it but only by altering the dns settings of the servers so that they could see each other without going through my router. It's a bit complicated to explain, I have a somewhat weird setup involving local dns servers, haproxy etc. But there's definitely something about dns and locally hosted servers that is causing the issue.

**Reply by chefblogger** [reply]
we will see - i just found out that yunhost has a new update for pixelfed available. I'm doing that right now and then I'll see if this strange follow problem still exists - a failed installation could always be responsible for such a bug
